src/renderers/shared/fiber/ReactFiberContext.js

```diff
@@ -129,13 +131,15 @@ exports.pushContextProvider = function(workInProgress : Fiber, didPerformWork :
     instance.__reactInternalMemoizedMergedChildContext = mergedContext;
   }
 
-  index++;
-  contextStack[index] = mergedContext;
-  didPerformWorkStack[index] = didPerformWork;
+  push(contextStackCursor, mergedContext, workInProgress);
+  push(didPerformWorkStackCursor, didPerformWork, workInProgress);
 };
 
 exports.resetContext = function() : void {
-  index = -1;
+  reset();
```

Comment:
We now have two calls to reset in a row because the only thing that calls this is:

https://github.com/facebook/react/blob/master/src/renderers/shared/fiber/ReactFiberScheduler.js#L157-L160

So we could just call `reset();` at the top of that function instead.